pymc3/tests/test_smc.py

```diff
@@ -47,6 +49,15 @@ def test_sample(self):
         mu1d = np.abs(x).mean(axis=0)
         np.testing.assert_allclose(self.muref, mu1d, rtol=0., atol=0.03)
 
+    @pytest.mark.xfail(condition=(theano.config.floatX == "float32"), reason="Fails on float32")
```

Comment:
How could this fail on float32, it's just running a model?

Reply:
Therei is a TypeError: expected type_num 11 (NPY_FLOAT32) got 12 error when evaluating the likelihood_logp function https://travis-ci.org/pymc-devs/pymc3/jobs/447528838#L1870

Reply:
You probably need to floatX the observed data.

Reply:
Right, thanks! This will fix it.